console/src/view/task.rs

```diff
@@ -123,13 +167,128 @@ impl TaskView {
+        frame.render_widget(percentiles_widget, histogram_area[0]);
+        frame.render_widget(percentiles_widget_1, percentiles_columns[0]);
+        frame.render_widget(percentiles_widget_2, percentiles_columns[1]);
+        frame.render_widget(histogram_sparkline, sparkline_area);
+    }
+}
+
+impl Details {
+    /// From the histogram, build a visual representation by trying to make as
+    // many buckets as the width of the render area.
+    fn make_chart_data(&self, width: u16) -> (Vec<u64>, HistogramMetadata) {
+        self.poll_times_histogram()
+            .map(|histogram| {
+                let step_size =
+                    ((histogram.max() - histogram.min()) as f64 / width as f64).ceil() as u64 + 1;
+                // `iter_linear` panics if step_size is 0
+                let data = if step_size > 0 {
+                    let mut found_first_nonzero = false;
+                    let data: Vec<u64> = histogram
+                        .iter_linear(step_size)
+                        .map(|it| it.count_since_last_iteration())
+                        .filter(|count| {
+                            // Remove the 0s from the leading side of the buckets.
+                            // Because HdrHistogram can return empty buckets depending
+                            // on its internal state, as it approximates values.
+                            if *count == 0 && !found_first_nonzero {
+                                false
+                            } else {
+                                found_first_nonzero = true;
+                                true
+                            }
+                        })
+                        .collect();
+                    data
+                } else {
+                    Vec::new()
+                };
+                let max_bucket = data.iter().max().copied().unwrap_or_default();
+                let min_bucket = data.iter().min().copied().unwrap_or_default();
+                (
+                    data,
+                    HistogramMetadata {
+                        max_value: histogram.max(),
+                        min_value: histogram.min(),
+                        max_bucket,
+                        min_bucket,
+                    },
+                )
+            })
+            .unwrap_or_default()
+    }
+
+    /// Get the important percentile values from the histogram and  make two paragraphs listing them
+    fn make_percentiles_widgets(&self, column_height: usize) -> (Text<'static>, Text<'static>) {
+        let percentiles_iter = self
+            .poll_times_histogram()
+            .map(|histogram| {
+                [10f64, 25f64, 50f64, 75f64, 90f64, 95f64, 99f64]
+                    .iter()
+                    .map(|i| (*i, histogram.value_at_percentile(*i)))
+                    .collect::<Vec<(f64, u64)>>()
+            })
+            .map(|pairs| {
+                pairs.into_iter().map(|pair| {
```

Comment:
nit: i *think* it should be possible to avoid the `collect`/`into_iter` here, and just have the first `map` closure return an iterator? that would be a bit more efficient, as we would be able to avoid allocating an intermediate `Vec`
```suggestion
            })
            .map(|pairs| {
                pairs.map(|pair| {
```